packages/core/src/shared/clipboard.ts

```diff
@@ -15,10 +15,40 @@
  */
 
 import { BaselineOffset, BooleanNumber } from '../types/enum';
-import type { IDocumentBody, ITextRun } from '../types/interfaces';
+import { CustomRangeType, type IDocumentBody, type ITextRun } from '../types/interfaces';
 import { Tools } from './tools';
 
 export function getBodySliceHtml(body: IDocumentBody, startIndex: number, endIndex: number) {
+    const { customRanges = [] } = body;
+    const customRangesInRange = customRanges.filter((range) => range.startIndex >= startIndex && range.endIndex <= endIndex);
+
+    let cursorIndex = startIndex;
+    let html = '';
+    customRangesInRange.forEach((range) => {
```

Comment:
Thinking about extensibility, what features should be builtin here? What about other features like mentions, extensions etc.?